docs/contributing/ui_guidelines.md

```diff
@@ -42,6 +43,30 @@ We use [Prettier](https://prettier.io/) for formatting and [ESLint](https://esli
 -   We follow a somewhat relaxed version of the [Airbnb styleguide](https://github.com/airbnb/javascript).
 -   Familiarise yourself with our [eslint-config-wagtail](https://github.com/wagtail/eslint-config-wagtail) configuration, which details our preferred code style.
 
+## Stimulus
+
+Wagtail uses [Stimulus](https://stimulus.hotwired.dev/) as a lightweight framework to attach interactive behaviour to DOM elements via `data-` attributes.
+
+### When to use Stimulus
+
+Stimulus is our [preferred library](https://github.com/wagtail/rfcs/pull/78) for simple client-side interactivity. It’s a good fit when:
+
+- The interactivity requires JavaScript. Otherwise, consider using HTML and CSS only.
+- Some of the logic is defined via HTML templates, not just JavaScript.
+- The interactivity is simple, and doesn’t require usage of more heavyweight libraries like React.
+
+Wagtail’s admin interface also leverages jQuery for similar scenarios. This is considered legacy and will eventually be removed. For new features, carefully consider whether existing jQuery code should be reused, or whether a rebuild with Stimulus is more appropriate.
+
+### How to build a Stimulus controller
+
+First think of how to name the controller. Keep it concise, one or two words ideally. Then,
+
+1. Start with the HTML templates, build as much of the UI as you can in HTML alone. Ensure it is accessible and follows the CSS guidelines.
+2. Create the controller file in our `client/src/controllers` folder, along with its tests (see [](testing)) and Storybook stories.
+3. For initialisation, consider which [controller lifecycle methods](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#methods) to use, if any (`connect`, `initialize`).
+4. If relevant, also consider how to handle the controlled element being removed from the DOM (`disconnect` lifecycle method).
+5. Document controller classes and methods with [JSDoc annotations](https://jsdoc.app/index.html).
+
```

Comment:
```suggestion

### Helpful tips

* Prefer controllers that do a small amount of 'work' that is collected together, instead of lots of large or specific controllers.
* Lean towards dispatching events for key behaviour in the UI interaction as this provides a great way for custom code to hook into this without an explicit API, but be sure to document these.
* Multiple controllers can be attached to one DOM element for composing behaviour, where practical split out behaviour to separate controllers.
* Consider when to document controller usage for non-contributors.
* When writing unit tests, note that DOM updates triggered by data attribute changes are completed async (next `microtick`) so will require a await Promise or similar to check for the changes in JSDom.
* Avoid hard-coding a controller's identifier, instead reference it with `this.identifier` if adjusting attributes. This way the controller can be used easily with a changed identifier or extended by other classes in the future.

```

Happy if we do not want these in v1.